handler/src/main/java/io/netty/handler/ssl/SslUtils.java

```diff
@@ -293,13 +298,22 @@ static int getEncryptedPacketLength(ByteBuf buffer, int offset) {
         if (tls) {
             // SSLv3 or TLS or GMSSLv1.0 or GMSSLv1.1 - Check ProtocolVersion
             int majorVersion = buffer.getUnsignedByte(offset + 1);
+            int version = buffer.getShort(offset + 1);
             if (majorVersion == 3 || buffer.getShort(offset + 1) == GMSSL_PROTOCOL_VERSION) {
                 // SSLv3 or TLS or GMSSLv1.0 or GMSSLv1.1
                 packetLength = unsignedShortBE(buffer, offset + 3) + SSL_RECORD_HEADER_LENGTH;
                 if (packetLength <= SSL_RECORD_HEADER_LENGTH) {
                     // Neither SSLv3 or TLSv1 (i.e. SSLv2 or bad data)
                     tls = false;
                 }
+            } else if (version == DTLS_1_0 || version == DTLS_1_2 || version == DTLS_1_3) {
+                // read each DTLS record's length until we can't anymore
+                while (buffer.readableBytes() >= offset + DTLS_RECORD_HEADER_LENGTH) {
```

Comment:
Why not return the first length and not loop ? I think this would be more expected.

Reply:
@normanmaurer I can double check today, but I am pretty sure the SSLHandler will end up just not processing those bytes. There is no outer loop that says "while there are unprocessed bytes do ...". I will double check though.

Reply:
This should be done by `ByteToMessageDecoder` under the hood

Reply:
👍 Ok. I will still double check. Getting slowed down today at work.